src/modules/rppt_tensor_effects_augmentations.cpp

```diff
@@ -505,6 +505,79 @@ RppStatus rppt_non_linear_blend_host(RppPtr_t srcPtr1,
     return RPP_SUCCESS;
 }
 
+/******************** ricap ********************/
+
+RppStatus rppt_ricap_host(RppPtr_t srcPtr,
+                          RpptDescPtr srcDescPtr,
+                          RppPtr_t dstPtr,
+                          RpptDescPtr dstDescPtr,
+                          Rpp32u *permutationTensor,
+                          RpptROIPtr roiPtrInputCropRegion,
+                          RpptRoiType roiType,
+                          rppHandle_t rppHandle)
+{
+    RppLayoutParams layoutParams = get_layout_params(srcDescPtr->layout, srcDescPtr->c);
+
+    if(srcDescPtr->n == 1) // BatchSize should always be greater than 1
```

Comment:
Does it work with batch size less than 4?

Reply:
yeah it works for any dataset with batchsize > 1